File: frontend/src/services/thought.service.ts
import { api } from '@/lib/api';
import { ApiResponse, PageResponse } from '@/types/common';
import { Thought, CreateThoughtRequest, PublicationStatus } from '@/types/content';

export const ThoughtService = {
    async getAll(page = 0, size = 10, status?: PublicationStatus): Promise<PageResponse<Thought>> {
        const params = new URLSearchParams({ page: page.toString(), size: size.toString() });
        if (status) params.append('status', status);

        const response = await api.get<any, ApiResponse<PageResponse<Thought>>>(`/thoughts?${params.toString()}`);
        return response.data;
    },

    async getById(id: number): Promise<Thought> {
        const response = await api.get<any, ApiResponse<Thought>>(`/thoughts/${id}`);
        return response.data;
    },

    async create(data: CreateThoughtRequest): Promise<Thought> {
        const response = await api.post<any, ApiResponse<Thought>>('/thoughts', data);
        return response.data;
    },

    async updateStatus(id: number, status: PublicationStatus): Promise<Thought> {
        const response = await api.patch<any, ApiResponse<Thought>>(`/thoughts/${id}/status`, null, { params: { status } });
        return response.data;
    },

    async delete(id: number): Promise<void> {
        await api.delete<any, ApiResponse<void>>(`/thoughts/${id}`);
    },

    async incrementView(id: number): Promise<void> {
        await api.post<any, ApiResponse<void>>(`/thoughts/${id}/view`);
    }
};

File: after-midnight-backend/src/main/java/com/aftermidnight/service/ThoughtService.java
package com.aftermidnight.service;

import com.aftermidnight.entity.PublicationStatus;
import com.aftermidnight.entity.Thought;
import com.aftermidnight.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface ThoughtService {
    Thought create(Thought thought);
    Page<Thought> getAllByUser(User user, Pageable pageable);
    Page<Thought> getAllByUserAndStatus(User user, PublicationStatus status, Pageable pageable);
    Page<Thought> getAllPublished(Pageable pageable);
    Optional<Thought> getById(Long id);
    void delete(Long id);
    Thought updateStatus(Long id, PublicationStatus status);
    void incrementViewCount(Long id);
}
